test/arbcall-test.jl

```diff
@@ -0,0 +1,130 @@
+        @test Arblib.jltype(arg) == type
+        @test Arblib.ctype(arg) == ctype
+    end
+
+    # Unsupported types
+    for str in ("fmpz_t x",
+                "fmpq_t x",
+                "mag_ptr res",
+                "arb_ptr res",
+                "acb_ptr res",
+                "const fmpz_t x",
+                "const fmpq_t x",
+                "mag_srcptr res",
+                "arb_srcptr res",
+                "acb_srcptr res",
+
+                # Internal types
+                "mpfr_rnd_t rnd",
+                "mp_limb_t lo",
+                "mp_bitcnt_t r",
+                "mp_ptr ycos",
+                "mp_srcptr x",
+                )
+        @test_throws KeyError arg = Arblib.Carg(str)
+    end
+
+    # Parsed incorrectly
+    for str in ("const char * inp",
+                )
+        @test_throws KeyError arg = Arblib.Carg(str)
+    end
+
+
+    # Parse errors
+    for str in (# Internal types
+                "mp_limb_t * error",
+                "mp_bitcnt_t * Qexp",
+                )
+        @test_throws ArgumentError arg = Arblib.Carg(str)
+    end
+end
+
+@testset "jlfname" begin
+    for (arbfname, name) in (# Supported types
+                             ("mag_set", :set),
+                             ("mag_set_d", :set),
+                             ("mag_set_ui", :set),
+                             ("arf_set", :set),
+                             ("arf_set_ui", :set),
+                             ("arf_set_si", :set),
+                             ("arf_set_d", :set),
+                             ("arb_set", :set),
+                             ("arb_set_arf", :set),
+                             ("arb_set_si", :set),
+                             ("arb_set_ui", :set),
+                             ("arb_set_d", :set),
+                             ("acb_set", :set),
+                             ("acb_set_ui", :set),
+                             ("acb_set_si", :set),
+                             ("acb_set_d", :set),
+                             ("acb_set_arb", :set),
+                             ("acb_set_si_si", :set),
+                             ("acb_set_d_d", :set),
+                             ("acb_set_arb_arb", :set),
+
+                             # Unsupported types
+                             ("arf_set_fmpz", :set_fmpz),
+                             ("arf_set_mpfr", :set_mpfr),
+                             ("acb_set_fmpq", :set_fmpq),
+                             ("arb_bin_uiui", :bin_uiui),
+
+                             # Deprecated types
+                             ("arf_set_mpz", :set_mpz),
+                             ("arf_set_fmpr", :set_fmpr),
+
+                             # Underscore methods
+                             ("_arb_vec_set", :_arb_vec_set),
+
+                             # Some special cases to be aware of and maybe change
+                             ("mag_set_d_lower", :set_d_lower),
+                             ("arb_ui_div", :ui_div),
```

Comment:
I'd have thought that this function would be named `arb_div_ui_arb`, but anyway:
We can either add `ui` to the list of prefixes, or keep it like it is.

When designing signature parsing we planned to allow providing julia function name, when heuristic for parsing the signature doesn't work: instead of using
`arbcall"void arb_ui_div(arb_t z, ulong x, const arb_t y, slong prec)"`
we'd need to use e.g.
```julia
@eval begin
   af = ArbFuncion("void arb_ui_div(arb_t z, ulong x, const arb_t y, slong prec)")
   jlcode(af, :div)
end
```
or  add 
```julia
macro arbcall(jlf_name::Symbol, signature_str)
    af = ArbFunction(signature_str)
    return jlcode(af, jlf_name)
end
```
for this purpose.